ownCloud File Provider/FileProviderServiceSource.m

```diff
@@ -81,27 +95,42 @@ - (BOOL)listener:(NSXPCListener *)listener shouldAcceptNewConnection:(NSXPCConne
 		[weakConnection invalidate];
 	};
 
-	[self.fileProviderExtension.core scheduleInCoreQueue:^{
-		[newConnection resume];
+	OCLogDebug(@"FileProviderServiceSource[%p].shouldAcceptNewConnection[enter]", self.fileProviderExtension);
+
+	[OCCoreManager.sharedCoreManager requestCoreForBookmark:_bookmark setup:nil completionHandler:^(OCCore * _Nullable core, NSError * _Nullable error) {
+		self->_core = core;
+
+		[core scheduleInCoreQueue:^{
+			OCLogDebug(@"FileProviderServiceSource[%p].shouldAcceptNewConnection[done]", self.fileProviderExtension);
+			[newConnection resume];
+		}];
 	}];
 
+	OCLogDebug(@"FileProviderServiceSource[%p].shouldAcceptNewConnection[core]: %@", self.fileProviderExtension, self.fileProviderExtension.core);
+
 	return (YES);
 }
 
 #pragma mark - Service API
 - (nullable NSProgress *)importItemNamed:(nullable NSString *)newFileName at:(OCItem *)parentItem fromURL:(NSURL *)inputFileURL isSecurityScoped:(BOOL)isSecurityScoped importByCopying:(BOOL)importByCopying automaticConflictResolutionNameStyle:(OCCoreDuplicateNameStyle)nameStyle placeholderCompletionHandler:(void(^)(NSError * _Nullable error))completionHandler
 {
-	return ([_fileProviderExtension.core importItemNamed:newFileName
-							  at:parentItem
-						     fromURL:inputFileURL
-					    isSecurityScoped:isSecurityScoped
-						     options:@{
-							     OCCoreOptionImportByCopying : @(importByCopying),
-							     OCCoreOptionAutomaticConflictResolutionNameStyle : @(nameStyle)
-						     }
-				placeholderCompletionHandler:^(NSError * _Nullable error, OCItem * _Nullable item) {
+	return ([_core importItemNamed:newFileName
```

Comment:
Should we check here, if `_core` is not `nil`, otherwise return an error or could it not happen, that `_core` is not `nil`?

Reply:
That method is only called over the XPC connection, which only is resumed once a core could successfully be requested in https://github.com/owncloud/ios-app/blob/8ef29aba3683691f959ad158633d8bd0bd310e1d/ownCloud%20File%20Provider/FileProviderServiceSource.m#L105 - like you already thought, `_core` can't be `nil` here.